dvc/tree/s3.py

```diff
@@ -210,7 +210,7 @@ def getsize(self, path_info):
 
     def _list_paths(self, path_info, max_items=None):
         with self._get_bucket(path_info.bucket) as bucket:
-            obj_summaries = bucket.objects.filter(Prefix=path_info.path)
+            obj_summaries = bucket.objects.filter(Prefix=(path_info / "/").path)
```

Comment:
This fix won't be sufficient because `path_info` won't always be the path to a directory here.

S3 supports arbitrary prefixes, meaning `path_info` could be a path to something like `ab/c` where we want to retrieve the list of any key in the S3 bucket that starts with the string `ab/c`, (so it would match `ab/cdef0`, `ab/cdef1`, etc). In this case, adding the trailing slash will end up returning the wrong set of keys.

Reply:
Great point @pmrowla !